pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,419 @@
+package pod
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/duration"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	"k8s.io/kubernetes/pkg/util/node"
+
+	"volcano.sh/apis/pkg/apis/batch/v1alpha1"
+	"volcano.sh/volcano/pkg/cli/util"
+)
+
+const (
+	// Name pod name
+	Name string = "Name"
+	// Ready pod ready
+	Ready string = "Ready"
+	// Status pod status
+	Status string = "Status"
+	// Restart pod restart
+	Restart string = "Restart"
+	// Age pod age
+	Age string = "Age"
+)
+
+type listFlags struct {
+	util.CommonFlags
+	// Namespace pod namespace
+	Namespace string
+	// JobName represents the pod created under this vcjob,
+	// filtered by volcano.sh/job-name label
+	// the default value is empty, which means
+	// that all pods under vcjob will be obtained.
+	JobName string
+	// allNamespace represents getting all namespaces
+	allNamespace bool
+	// QueueName represents queue name
+	QueueName string
+}
+
+var listPodFlags = &listFlags{}
+
+// InitListFlags init list command flags.
+func InitListFlags(cmd *cobra.Command) {
+	util.InitFlags(cmd, &listPodFlags.CommonFlags)
+
+	cmd.Flags().StringVarP(&listPodFlags.QueueName, "queue", "q", "", "list pod with specified queue name")
+	cmd.Flags().StringVarP(&listPodFlags.JobName, "job", "j", "", "list pod with specified job name")
```

Comment:
What if user not specified job name? Does all pods in a namespace will be printed whether it's a vcjob?

Reply:
We should clarify the definition of this feature. If we only allow list pods under a vcjob, then not specifying the job name is an abnormal scenario, and some improvements should be given to users: )

Reply:
```go
// JobNameKey job name key used in pod annotation / labels
JobNameKey = "volcano.sh/job-name"
```

If no parameters are entered, by default only pods created by vcjob will be obtained, which will be filtered out by the selector